sda/treenode.go

```diff
@@ -379,15 +383,20 @@ func (n *TreeNodeInstance) dispatchMsgToProtocol(sdaMsg *ProtocolMsg) error {
 		log.Lvl3(n.Name(), "Not done aggregating children msgs")
 		return nil
 	}
-	log.Lvlf5("TNI dispatching -Message is: %+v", sdaMsg.Msg)
+	log.LLvlf5("%s->%s: Message is: %+v", sdaMsg.ServerIdentity.First(),
+		n.Name(), sdaMsg.Msg)
 
 	switch {
 	case n.channels[msgType] != nil:
-		log.Lvl4(n.Info(), "Dispatching to channel")
+		log.LLvl4(n.Name(), "Dispatching to channel")
 		err = n.DispatchChannel(msgs)
 	case n.handlers[msgType] != nil:
 		log.Lvl4("Dispatching to handler", n.ServerIdentity().Addresses)
-		err = n.dispatchHandler(msgs)
+		go func() {
```

Comment:
This is needed because `bftcosi` does wait in a handler on a channel to be filled with a value. But this value will be sent from another handler. So either we don't allow for more than one handler to be active, or we accept this go.